From 6fe494ec5b0baf2fa604f5ae6a6237eb5dc0b66a Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Mon, 9 Mar 2026 13:55:01 +0100
Subject: [PATCH] [codec,clear] fix destination checks

Backport of commit 7d8fdce2d0ef337cb86cb37fc0c436c905e04d77.

Made-with: Cursor
---
 libfreerdp/codec/clear.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/libfreerdp/codec/clear.c b/libfreerdp/codec/clear.c
index 2a4f894ea..4c42bb2bf 100644
--- a/libfreerdp/codec/clear.c
+++ b/libfreerdp/codec/clear.c
@@ -490,16 +490,16 @@ static BOOL clear_decompress_subcodecs_data(CLEAR_CONTEXT* WINPR_RESTRICT clear,
 		nXDstRel = nXDst + xStart;
 		nYDstRel = nYDst + yStart;
 
-		if (1ull * xStart + width > nWidth)
+		if (1ull * nXDstRel + width > nDstWidth)
 		{
-			WLog_ERR(TAG, "xStart %" PRIu16 " + width %" PRIu16 " > nWidth %" PRIu32 "", xStart,
-			         width, nWidth);
+			WLog_ERR(TAG, "nXDstRel %" PRIu32 " + width %" PRIu16 " > nDstWidth %" PRIu32 "",
+			         nXDstRel, width, nDstWidth);
 			return FALSE;
 		}
-		if (1ull * yStart + height > nHeight)
+		if (1ull * nYDstRel + height > nDstHeight)
 		{
-			WLog_ERR(TAG, "yStart %" PRIu16 " + height %" PRIu16 " > nHeight %" PRIu32 "", yStart,
-			         height, nHeight);
+			WLog_ERR(TAG, "nYDstRel %" PRIu32 " + height %" PRIu16 " > nDstHeight %" PRIu32 "",
+			         nYDstRel, height, nDstHeight);
 			return FALSE;
 		}
 
-- 
2.53.0

